### PR TITLE
Publish 2024-04-15-preview version of Batch Avatar API spec

### PR DESCRIPTION
**Pull request by zhifzhan**
# Data Plane API - Pull Request

This PR is created based on private repo PR https://github.com/Azure/azure-rest-api-specs-pr/pull/15489 which has been reviewed and approved.

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue: https://github.com/Azure/azure-rest-api-specs-pr/issues/14999

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>

* Design Document:
* Previous API Spec Doc:
* Updated paths:

Compared to https://github.com/Azure/azure-rest-api-specs-pr/pull/15489, the following changes are made:
- API path changed from `/talkingavatar/` to `/avatar/` to be align with our public product name.
- Renamed properties in `BatchSynthesisBillingDetails`
  - `neural` to `neuralCharacters`
  - `customNeural` to `customNeuralCharacters`
- Two int properties were added in `BatchSynthesisBillingDetails`
  - `aoaiCharacters`: integer
  - `aoaiHDCharacters`: integer
- Removed unused properties from `BatchSynthesisProperties`
  - `concatenateResult`: boolean
  - `decompressOutputFiles`: boolean

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Schedule a data plane REST API spec review](https://aka.ms/azsdk/onboarding/restapischedule)

</details>

<details>
  <summary>Checks stuck in `queued` state?</summary>
If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.  
</details>

### CODE REVIEW

**Comment by mikekistler** [comment]
Here's a link to the notes of the API review we did last Oct:

https://github.com/Azure/azure-rest-api-specs-pr/issues/14999#issuecomment-1771864199

Conclusion there was this is good for preview with a few small tweaks, so I'll approve.

**Review by mikekistler — APPROVED** [review]
Looks good for preview. 👍 

We still need to resolve the unified versioning across the whole service before this moves to GA.

File: specification/cognitiveservices/data-plane/Speech/BatchAvatar/preview/2024-04-15-preview/batchavatar.json [hunk 1]
```diff
@@ -0,0 +1,846 @@
...
+      "enum": [
+        "PlainText",
+        "SSML"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "InputKind",
+        "modelAsString": true
+      }
+    }
```
Comment (weidongxu-microsoft) [inline specification/cognitiveservices/data-plane/Speech/BatchAvatar/preview/2024-04-15-preview/batchavatar.json:763]: nit for future, Speech may want to put the common definitions to a single place.

I remember a `InputKind` at https://github.com/Azure/azure-rest-api-specs/pull/28272/files#r1542276805

File: specification/cognitiveservices/data-plane/Speech/BatchAvatar/preview/2024-04-15-preview/batchavatar.json [hunk 2]
```diff
@@ -0,0 +1,846 @@
...
+    "QueryMaxPageSize": {
+      "name": "maxpagesize",
+      "in": "query",
+      "description": "The maximum number of result items to include in a single response.",
+      "required": false,
+      "type": "integer",
+      "format": "int32",
+      "x-ms-parameter-location": "method"
+    },
+    "PathId": {
```
Comment (weidongxu-microsoft) [inline specification/cognitiveservices/data-plane/Speech/BatchAvatar/preview/2024-04-15-preview/batchavatar.json:834]: nit, do we have a better name?

It probably won't show up in SDK or doc, but `PathId` seems not correct. Maybe `BatchSynthesisId` or `AvatarId` (whichever appropriate for your API)?

File: specification/cognitiveservices/data-plane/Speech/BatchAvatar/preview/2024-04-15-preview/batchavatar.json [hunk 3]
```diff
@@ -0,0 +1,846 @@
...
+        },
+        "lastActionDateTime": {
+          "description": "The time-stamp when the current status was entered.",
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        },
+        "id": {
+          "$ref": "#/definitions/ResourceId"
+        },
```
Comment (weidongxu-microsoft) [inline specification/cognitiveservices/data-plane/Speech/BatchAvatar/preview/2024-04-15-preview/batchavatar.json:334]: Is this the `{id}` on path? Should it be readOnly in schema?

File: specification/cognitiveservices/data-plane/Speech/BatchAvatar/preview/2024-04-15-preview/batchavatar.json [hunk 4]
```diff
@@ -0,0 +1,846 @@
...
+      ],
+      "type": "object",
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/Error"
+        }
+      }
+    },
+    "InnerError": {
+      "description": "New Inner Error format which conforms to Cognitive Services API Guidelines which is available at https://microsoft.sharepoint.com/%3Aw%3A/t/CognitiveServicesPMO/EUoytcrjuJdKpeOKIK_QRC8BPtUYQpKBi8JsWyeDMRsWlQ?e=CPq8ow. This contains required properties ErrorCode, message and optional properties target, inner error(this can be nested).",
```
Comment (weidongxu-microsoft) [inline specification/cognitiveservices/data-plane/Speech/BatchAvatar/preview/2024-04-15-preview/batchavatar.json:656]: Does people outside of microsoft able to access this sharepoint URL?

File: specification/cognitiveservices/data-plane/Speech/BatchAvatar/preview/2024-04-15-preview/batchavatar.json [hunk 5]
```diff
@@ -0,0 +1,846 @@
...
+          "type": "string"
+        }
+      }
+    },
+    "BatchSynthesisProperties": {
+      "description": "Detailed properties of batch synthesis task.",
+      "type": "object",
+      "properties": {
+        "timeToLiveInHours": {
+          "description": "How long the batch task will be kept in the system after it has completed. Once the batch task reaches the time to live after completion (succeeded or failed) it will be automatically deleted. The longest supported duration is 744 hours (31 days). The default value is 744 hours. ",
```
Comment (weidongxu-microsoft) [inline specification/cognitiveservices/data-plane/Speech/BatchAvatar/preview/2024-04-15-preview/batchavatar.json:455]: In additional to decsription, default value better to be put to e.g. `"default": 774`

**Review by weidongxu-microsoft — APPROVED** [review]
LGTM

One question https://github.com/Azure/azure-rest-api-specs/pull/28279#discussion_r1562018942

File: specification/cognitiveservices/data-plane/Speech/BatchAvatar/preview/2024-04-15-preview/batchavatar.json (hunk 3 above)
Comment (zhifzhan) [reply specification/cognitiveservices/data-plane/Speech/BatchAvatar/preview/2024-04-15-preview/batchavatar.json]: Ah yes it is the id on path. We don't allow user to change it in payload. Updated and marked as readOnly.

File: specification/cognitiveservices/data-plane/Speech/BatchAvatar/preview/2024-04-15-preview/batchavatar.json (hunk 2 above)
Comment (zhifzhan) [reply specification/cognitiveservices/data-plane/Speech/BatchAvatar/preview/2024-04-15-preview/batchavatar.json]: Thanks, updated to `SynthesisId`

File: specification/cognitiveservices/data-plane/Speech/BatchAvatar/preview/2024-04-15-preview/batchavatar.json (hunk 5 above)
Comment (zhifzhan) [reply specification/cognitiveservices/data-plane/Speech/BatchAvatar/preview/2024-04-15-preview/batchavatar.json:456]: Thanks! Added `"default"`

File: specification/cognitiveservices/data-plane/Speech/BatchAvatar/preview/2024-04-15-preview/batchavatar.json (hunk 1 above)
Comment (zhifzhan) [reply specification/cognitiveservices/data-plane/Speech/BatchAvatar/preview/2024-04-15-preview/batchavatar.json:765]: Yes they are born from the same structure.
For now we are not sure how they may evolve in future yet: They may go differently to have different supported `inputKind`.
But yeah we'll consider merging these shared things in future version.

**Review by raosuhas — APPROVED** [review]

**Pull request merged**